lptree.pas

```diff
@@ -2719,6 +2728,95 @@ function TLapeTree_InternalMethod_New.Compile(var Offset: Integer): TResVar;
   Param.Spill(1);
 end;
 
+constructor TLapeTree_InternalMethod_Operator.Create(AOperator:EOperator; ACompiler: TLapeCompilerBase; ADocPos: PDocPos = nil);
+begin
+  inherited Create(ACompiler, ADocPos);
+  if AOperator = op_UnaryMinus then 
+    self.FOperator := op_Minus
+  else
+    self.FOperator := AOperator;
+end;
+
+function TLapeTree_InternalMethod_Operator.resType: TLapeType;
+var
+  _LapeOperator: TLapeGlobalVar;
+  typ:TLapeType;
+  methodParams:TLapeTypeArray;
+  methodVar,identVar: TLapeTree_GlobalVar;
+  i:Int32;
+begin
+  (*
+    Might be ideal to find a faster way to look up the restype of overloaded ops.
+    This might involve a bit of work.
+    Currently this is fine, as I see no negative affect on compile-time.
+  *)
```

Comment:
You can use setExpr to set the overloaded method and let TLapeTree_Invoke take care of this. For example, Write/WriteLn/Assert are internal methods that let overloaded handling be done by _Invoke.